flux/src/logger.rs

```diff
@@ -1,23 +1,85 @@
-use std::{fs, io, sync::Arc};
+use std::{
+    fs,
+    io::{self, Write},
+    sync::Arc,
+};
 
 use flux_common::config::CONFIG;
-use tracing::Level;
+use tracing::{Dispatch, Level};
 use tracing_subscriber::{filter::Targets, fmt::writer::BoxMakeWriter, prelude::*, Registry};
+use tracing_timing::TimingLayer;
 
-pub fn install() -> io::Result<()> {
-    if CONFIG.dump_checker_trace {
-        let log_dir = &CONFIG.log_dir;
+const ONE_HUNDRED_MILLIS: u64 = 100_000_000;
+const ONE_MINUTE: u64 = 60_000_000_000;
+
+const CHECKER_FILE: &str = "checker";
+const TIMINGS_FILE: &str = "timings";
+
+pub fn install() -> io::Result<impl FnOnce() -> io::Result<()>> {
+    let log_dir = &CONFIG.log_dir;
+
+    if CONFIG.dump_checker_trace || CONFIG.dump_timings {
         fs::create_dir_all(log_dir)?;
-        let file = fs::File::create(log_dir.join("checker"))?;
+    }
+
+    let fmt_layer = if CONFIG.dump_checker_trace {
+        let file = fs::File::create(log_dir.join(CHECKER_FILE))?;
 
         let writer = BoxMakeWriter::new(Arc::new(file));
-        let fmt_layer = tracing_subscriber::fmt::layer()
-            .with_writer(writer)
-            .json()
-            .with_filter(Targets::new().with_target("flux_typeck::checker", Level::DEBUG));
+        Some(
+            tracing_subscriber::fmt::layer()
+                .with_writer(writer)
+                .json()
+                .with_filter(Targets::new().with_target("flux_typeck::checker", Level::DEBUG)),
```

Comment:
`Option<Layer>` behaves as a toggle for the layer, see [this example](https://docs.rs/tracing-subscriber/0.3.16/tracing_subscriber/layer/index.html#runtime-configuration-with-layers).

The types don't work otherwise.